javascriptlib/src/mill/javascriptlib/TestModule.scala

```diff
@@ -25,6 +25,82 @@ trait TestModule extends TaskModule {
 object TestModule {
   type TestResult = Unit
 
+  trait Coverage extends TypeScriptModule with TestModule {
+    override def npmDevDeps: T[Seq[String]] = Task {
+      super.npmDevDeps() ++ Seq("serve@14.2.4")
+    }
+
+    private[TestModule] def runCoverage: T[TestResult]
+
+    protected def coverageTask(args: Task[Seq[String]]): Task[TestResult] = Task { runCoverage() }
+
+    def coverage(args: String*): Command[TestResult] =
+      Task.Command {
+        coverageTask(Task.Anon { args })()
+      }
+
+    // <rootDir> = '/out'; allow coverage resolve distributed source files.
+    // & define coverage files relative to <rootDir>.
+    private[TestModule] def link: Task[Unit] = Task.Anon {
+      os.symlink(Task.workspace / "out/node_modules", npmInstall().path / "node_modules")
+      os.symlink(Task.workspace / "out/tsconfig.json", compile()._1.path / "tsconfig.json")
+      if (os.exists(compile()._1.path / ".nycrc"))
+        os.symlink(Task.workspace / "out/.nycrc", compile()._1.path / ".nycrc")
+    }
+
+    def nycrcBuilder: Task[Path] = Task.Anon {
```

Comment:
This task is very weird. It seems like it should be a `T[PathRef]` returning the config file it writes

Reply:
Also what's `nycrc`? Maybe use a longer name or add a comment?

Reply:
https://www.npmjs.com/package/nyc

Reply:
nyc does not have a full name, renamed it to `istanbulNycrcConfigBuilder`